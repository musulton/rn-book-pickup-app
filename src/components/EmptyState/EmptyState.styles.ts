import {StyleSheet, ViewStyle} from "react-native";

interface Styles {
    emptyState: ViewStyle
}

const styles = StyleSheet.create<Styles>({
    emptyState: {
        display: "flex",
        flex: 1,
        justifyContent: "center",
        alignItems: "center"
    }
});

export default styles
