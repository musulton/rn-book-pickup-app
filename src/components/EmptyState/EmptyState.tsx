import {Text, View} from "react-native";
import React from "react";

import styles from "./EmptyState.styles";

interface EmptyStateProps {
    text?: string
}

const EmptyState: React.FC<EmptyStateProps> = (props) => (
    <View style={styles.emptyState}>
        <Text>{props.text || "Data is nof found"}</Text>
    </View>
)

export default EmptyState
