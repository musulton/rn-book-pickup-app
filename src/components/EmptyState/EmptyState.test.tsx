import {render, screen} from "@testing-library/react-native";

import EmptyState from "./EmptyState";

describe("<EmptyState />", () => {
    it("should render correctly with props", () => {
        const props = {
            text: "Test",
        }

        render(<EmptyState {...props} />)

        expect(screen.toJSON()).toMatchSnapshot();
    })

    it("should render correctly without props", () => {
        render(<EmptyState />)

        expect(screen.toJSON()).toMatchSnapshot();
    })
});
