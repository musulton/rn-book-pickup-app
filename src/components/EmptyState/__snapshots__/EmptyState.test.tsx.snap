// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should render correctly with props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Test
  </Text>
</View>
`;

exports[`<EmptyState /> should render correctly without props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Data is nof found
  </Text>
</View>
`;
